test/utils/noderesourcetopologies/noderesourcetopologies.go

```diff
@@ -321,14 +322,44 @@ func AccumulateNames(nrts []nrtv1alpha2.NodeResourceTopology) sets.String {
 	return nodeNames
 }
 
-func FilterTopologyManagerPolicy(nrts []nrtv1alpha2.NodeResourceTopology, tmPolicy nrtv1alpha2.TopologyManagerPolicy) []nrtv1alpha2.NodeResourceTopology {
+func FilterByTopologyManagerPolicy(nrts []nrtv1alpha2.NodeResourceTopology, tmPolicy string) []nrtv1alpha2.NodeResourceTopology {
+	return filterTopologyManagerPolicy(nrts, tmPolicy, "")
+}
+
+func FilterByTopologyManagerScope(nrts []nrtv1alpha2.NodeResourceTopology, tmScope string) []nrtv1alpha2.NodeResourceTopology {
+	return filterTopologyManagerPolicy(nrts, "", tmScope)
+}
+func FilterByTopologyManagerPolicyAndScope(nrts []nrtv1alpha2.NodeResourceTopology, tmPolicy, tmScope string) []nrtv1alpha2.NodeResourceTopology {
+	return filterTopologyManagerPolicy(nrts, tmPolicy, tmScope)
+}
+
+func filterTopologyManagerPolicy(nrts []nrtv1alpha2.NodeResourceTopology, tmPolicy, tmScope string) []nrtv1alpha2.NodeResourceTopology {
```

Comment:
I'd split the function in 2 filters: 1 by scope, 1 by policy and compose them by chaining in `FilterByTopologyManager PolicyAndScope` (yes, this will duplicate a `for` loop, and is not a big deal).